main-command/src/main/scala/sbt/ServerHandler.scala

```diff
@@ -0,0 +1,67 @@
+/*
+ * sbt
+ * Copyright 2011 - 2017, Lightbend, Inc.
+ * Copyright 2008 - 2010, Mark Harrah
+ * Licensed under BSD-3-Clause license (see LICENSE)
+ */
+
+package sbt
+
```

Comment:
could you make these `sbt.internal.server` please so we can break their API, please?

Reply:
If we are exposing this out to plugins, then it needs to be public, no?
Are we saying that people can extend it, but the extension is not guaranteed to work down the line?

Reply:
yes, it's exposed with 0 compatibility promises. caveat emptor

Reply:
I think it's fine if it's presented as an experimental/beta functionality.